core/dbt/include/global_project/macros/utils/data_types.sql

```diff
@@ -0,0 +1,117 @@
+{# string  -------------------------------------------------     #}
+
+{%- macro type_string() -%}
+  {{ return(adapter.dispatch('type_string', 'dbt')()) }}
+{%- endmacro -%}
+
+{% macro default__type_string() %}
+    {{ return(api.Column.translate_type("string")) }}
+{% endmacro %}
+
+-- This will return 'text' by default
+-- On Postgres + Snowflake, that's equivalent to varchar (no size)
+-- Redshift will treat that as varchar(256)
```

Comment:
I'm leaving these comments in, unless anyone feels strongly that they should be removed before merging. (They can always be removed after merging.)